pkg/services/ngalert/api/api_alertmanager.go

```diff
@@ -129,6 +130,19 @@ func (srv AlertmanagerSrv) RouteCreateSilence(c *models.ReqContext, postableSile
 		return ErrResp(http.StatusForbidden, errors.New("permission denied"), "")
 	}
 
+	// Validate our silence to avoid panics
+	//
+	// Question for reviewers: do we care about the format registry? My reading
+	// of the code is that an empty format registry means "all data validates,
+	// but required fields must be present." Since we didn't evaluate before,
+	// that's the existing behavior. If we want an empty registry, I think we can
+	// just create it on each API call - it's not a very high-throughput API.
+	err := postableSilence.Validate(strfmt.NewFormats())
```

Comment:
I tried to look at the Alertmanager API to see what is expected there and its quite hard to follow the code. It could be something like `strfmt.Default`?

Reply:
I think that's equivalent to what I have here, without creating an object each time. Done.